### PR TITLE
add 'convertImgsToVideo'

### PR DESCRIPTION
**Pull request by uniray7**

### CODE REVIEW

File: src/media.js [hunk 1]
```diff
@@ -372,9 +373,40 @@ var deleteMediaImages = function(job) {
...
-
-
+    var mediaObj = JSON.parse(job.payload);
+    var keyPrefix = mediaObj.content.shardingKey+'/media/'+mediaObj.sid+'/';
+    if (mediaObj.type === 'livePhoto'){
+      keyPrefix += 'live/';
+    } 
+    keyPrefix = keyPrefix + mediaObj.content.quality[0] + '/';  
+    //var storeUrl = mediaObj.content.storeUrl;  
+    var storeUrl = 'http://192.168.1.27:6559/';
```
Comment (weitingchou) [inline src/media.js]: The storeUrl is hardcoded.

File: src/media.js [hunk 2]
```diff
@@ -372,9 +373,40 @@ var deleteMediaImages = function(job) {
...
+    } 
+    keyPrefix = keyPrefix + mediaObj.content.quality[0] + '/';  
+    //var storeUrl = mediaObj.content.storeUrl;  
+    var storeUrl = 'http://192.168.1.27:6559/';  
+    ffmpeg()
+    .input( storeUrl+keyPrefix+'%d.jpg' )
+    .inputFPS(25)
+    .fps(25)
+    .on('end', function() {
+      console.log('video has been converted successfully');
```
Comment (weitingchou) [inline src/media.js]: Should move this message to debug level.

File: src/media.js [hunk 3]
```diff
@@ -372,9 +373,40 @@ var deleteMediaImages = function(job) {
...
+      fs.readFile(mediaObj.sid+'.mp4', function(err, data){
+        if(err){ return callback(err);}
```
Comment (weitingchou) [inline src/media.js]: Where dose this callback come from? It should be reportException. And  should leave a space between "(err){ return".

File: src/media.js [hunk 4]
```diff
@@ -372,9 +373,40 @@ var deleteMediaImages = function(job) {
...
+    .on('end', function() {
+      console.log('video has been converted successfully');
+      fs.readFile(mediaObj.sid+'.mp4', function(err, data){
+        if(err){ return callback(err);}  
+        var keyArr = [ mediaObj.content.shardingKey, 'media', mediaObj.sid, 'live', 'video.mp4' ];
+        store.create(keyArr, data, function(err, result) {
+          if (err) { return callback(err); }
+          fs.unlink(mediaObj.sid+'.mp4', function(err, data){
+            if (err) { return callback(err); }
+              job.workComplete(JSON.stringify({
```
Comment (weitingchou) [inline src/media.js]: Should have the same indents as the above line.

File: package.json [hunk 5]
```diff
@@ -18,6 +18,7 @@
     "gm": "1.21.1",
     "lodash": "4.12.0",
     "moment": "2.13.0",
+    "randomstring": "^1.1.5",
```
Comment (weitingchou) [inline package.json]: Should use the exact version number.

File: src/media.js (hunk 1 above)
Comment (uniray7) [reply src/media.js]: is fixed in later commit

File: src/media.js (hunk 2 above)
Comment (uniray7) [reply src/media.js]: is fixed in later commit

**Pull request merged**